src/createReduxForm.js

```diff
@@ -646,16 +648,19 @@ const createReduxForm = (structure: Structure<*, *>) => {
               ? values
               : setIn(values, name, value)
             const syncValidationPasses = submitting || !getIn(syncErrors, name)
-            const isBlurredField =
+            const fieldNeedsValidation =
               !submitting &&
-              (!asyncBlurFields ||
-                ~asyncBlurFields.indexOf(name.replace(/\[[0-9]+\]/g, '[]')))
+              trigger === 'blur'
+                ? (!asyncBlurFields ||
+                  ~asyncBlurFields.indexOf(name.replace(/\[[0-9]+\]/g, '[]')))
+                : (!asyncChangeFields ||
```

Comment:
asyncBlurFields: ['username', 'email']
asyncChangeFields: []

when submitting this form - name is undefined,
so error Cannot read property 'replace' of undefined is thrown

Reply:
I'm not able to recreate this. The `name` variable is ultimately coming from [here](https://github.com/ericbiewener/redux-form/blob/270ea7b1114ff57cbb74e3bdf0faac4ccc575b3f/src/ConnectedField.js?utf8=%E2%9C%93#L94), which is the `name` prop you have given to your `Field`. Are you sure you have that prop on your field?

Reply:
@ericbiewener I am seeing something similar. Looking at the backtrace, I see that `asyncValidate` is called without arguments:

https://github.com/erikras/redux-form/blob/f23a36efb81bc18e2f98f155a4924084539d7fc6/src/handleSubmit.js#L95

However it expects `name` (and other arguments):

https://github.com/erikras/redux-form/blob/c1fc3ef5e9e7daf7ca66712109296ce1630fc1c1/src/createReduxForm.js#L652-L656

Reply:
Ah, now I see the bug. I looks like this code [has been modified](https://github.com/erikras/redux-form/pull/3647) a bit since this PR was merged, but I expect the issue to still exist. Let's [continue this conversation on that PR](https://github.com/erikras/redux-form/pull/3647/files#r154708730) since it contains the latest code.